### ISSUE TITLE
Java: CommentsIndentation false positive for comments in multiline arguments

### ISSUE THREAD

**Issue by christianhujer** [checkstyle/checkstyle#8010]
https://checkstyle.org/config_misc.html#CommentsIndentation

```

D:\checkstyletest>type config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://checkstyle.sourceforge.net/dtds/configuration_1_3.dtd">

<module name = "Checker">
    <module name="TreeWalker">
        <module name="CommentsIndentation"/>
    </module>
</module>

D:\checkstyletest>type test\*.java 
class Point {
    @SuppressWarnings({"checkstyle:ParenPad", "checkstyle:CommentIndentation"})
    private static final Set<Point> neighborSet = of(
            //@formatter:off
            P(-1,  1), P( 0,  1), P( 1,  1),
            P(-1,  0),            P( 1,  0),
            P(-1, -1), P( 0, -1), P( 1, -1)
            //@formatter:on   this is line 8 with unexpected violation
    );

    // …
}
D:\checkstyletest>java -jar -Duser.language=en -Duser.country=US \
     checkstyle-8.31-all.jar -c config.xml test\*.java 
Starting audit...
[ERROR] Test.java:8: Comment has incorrect indentation level 12, 
      expected is 4, indentation should be the same level as line 9. 
      [CommentsIndentation]
Audit done.
```

Checkstyle reports that the comment `//@formatter:on` is at a wrong level of indentation, which it isn't.

Expected behavior: comments within round parentheses are expected to be indented the same way as other lines within the round parentheses.

Actual behavior: Checkstyle reports a warning `CommentIndentation` about the comment `//@formatter:on`.

This may be related to #2998 and, from a user perspective, maybe a modification of the fix for #2998 will fix this one as well.

**Comment by wltan** [comment]
Before an admin complains about it, here is a sample run from the issue template:
```

D:\checkstyletest>type config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://checkstyle.sourceforge.net/dtds/configuration_1_3.dtd">

<module name = "Checker">
    <module name="TreeWalker">
        <module name="CommentsIndentation"/>
    </module>
</module>

D:\checkstyletest>type test\*.java 
class Point {
    @SuppressWarnings({"checkstyle:ParenPad", "checkstyle:CommentIndentation"})
    private static final Set<Point> neighborSet = of(
            //@formatter:off
            P(-1,  1), P( 0,  1), P( 1,  1),
            P(-1,  0),            P( 1,  0),
            P(-1, -1), P( 0, -1), P( 1, -1)
            //@formatter:on
    );

    // …
}
D:\checkstyletest>java -jar -Duser.language=en -Duser.country=US checkstyle-8.31-all.jar -c config.xml test\*.java 
Starting audit...
[ERROR] D:\checkstyletest\test\Test.java:8: Comment has incorrect indentation level 12, expected is 4, indentation should be the same level as line 9. [CommentsIndentation]
Audit done.
```
Agreed that it seems to be related to #2998.

**Comment by rnveach** [comment]
Violation is with the comment on the following lines:
````
            P(-1, -1), P( 0, -1), P( 1, -1)
            //@formatter:on
    );
````

Checkstyle says violation should be same indentation as `);` and it won't accept it being at the same spot as `P(-1, -1)`.

The following are similar examples from our descriptions at https://checkstyle.org/config_misc.html#CommentsIndentation_Description :
Example 2, 4, 7.

To my eyes, comment looks in a good position.

@romani Please finalize. I am for approving issue.

**Comment by AkMo3** [comment]
I am on it.

**Comment by rnveach** [comment]
Fix was merged
